src/parsers/manifest/dash/indexes/get_index_segment.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ISegment } from "../../../../manifest";
+
+/**
+ * Construct index segment for the given index.
+ * Returns null if the index segment is on the init segment.
+ * @param {Object} index
+ * @returns {Object}
+ */
+export default function getIndexSegment(
+  index: { timescale: number;
+           representationIndex?: { media: string;
+                                  range?: [number, number]; } |
+                                 null;
+           indexRange?: [number, number];
+           indexTimeOffset : number; }
+) : ISegment|null {
+  const { representationIndex } = index;
+  if (representationIndex == null) {
+    return null;
+  }
+  return { id: "representationIndex",
+           isInit: false,
+           privateInfos: { indexOnly: true },
```

Comment:
Disturbing to always rely on `privateInfos` for a public concept of `indexSegment`